lib/any.js

```diff
@@ -519,7 +521,7 @@ internals.Any.prototype._validate = function (value, state, options, reference)
                     errors.push(Errors.create('any.default', defaultValue.error, state, options));
                 }
             }
-            else {
+            else if (!options.noDefaults) {
```

Comment:
@Marsup you are right, i was looking for adding the minimum of complexity in this area, since there are plenty of conditions out there.

Let me know if it would be better to change each conditions to ensure that the "noDefaults" option is not set to "false".

Thanks !